crates/comm/src/peer/mod.rs

```diff
@@ -37,6 +40,32 @@ pub const CHANNEL_SIZE: usize = 32;
 /// [SocketPacket]s.
 const MAX_PROTOCOL_PACKET_CHUNK_SIZE: usize = UDP_MAX_DATAGRAM_SIZE - MIN_SOCKET_PACKET_SIZE;
 
+/// An enumeration of possible errors that can occur when working with peers.
+#[derive(Error, Debug)]
+pub enum PeerError {
+    // Failed to send packet between threads.
+    #[error("Failed to send packet to network thread")]
+    NetworkSendFail(#[from] SendError<SocketPacket>),
+    // Failed to send packet between threads.
+    #[error("Failed to send packet to application")]
+    ApplicationSendFail(#[from] SendError<ProtocolPacket>),
+    // Failed to decode decrypted packet
+    #[error("Failed to decode decrypted packet")]
+    DecodeFail(#[from] PacketDecodeError),
+    // Failed to encode packet for encryption
+    #[error("Failed to encode packet for encryption")]
+    EncodeFail(#[from] PacketEncodeError),
+    // Failure in double ratchet
+    #[error("Failure in double ratchet")]
+    DRFail(#[from] DoubleRatchetError),
+    // Generic error with signature
+    #[error("Failure in signature verification")]
+    SigFail(#[from] SigningError),
+    /// The packet we received does not conform to some format
+    #[error("Bad packet")]
+    BadPacket,
+}
+
```

Comment:
Any particular reason for moving `PeerError` about? It will make merging harder.

Reply:
It could probably go in `peer/error.rs` to make this a little easier once it's merged.

Reply:
good idea (moving it to `peer/error.rs`.